core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -186,7 +186,7 @@ V get()
             this.wait();
           }
           catch (InterruptedException e) {
-            Thread.interrupted();
```

Comment:
Perhaps we should prohibit this method on the forbidden-apis level?

Reply:
I agree! One can call `Thread.interrupted()` within an `if` check or assign it to a variable. Should we prohibit to assigning a variable too or not?

    <module name="Regexp">
      <property name="format" value='Thread.interrupted();'/>
      <property name="illegalPattern" value="true"/>
      <property name="message" value="Thread.interrupted(); shouldn't be used individually"/>
    </module>

Reply:
It should be prohibited via forbidden-apis, not Checkstyle.

> Should we prohibit to assigning a variable too or not?

Do you mean prohibiting `Thread#interrupt()`? It's called in a lot of places.

Reply:
I mean we do not forbid to use Thread.interrupted() i.e. these are valid usages:

      boolean needsToShutdown = Thread.interrupted() || shuttingDown;
      ...
      if (Thread.interrupted()) {
        throw new InterruptedException();
      }

It can be assigned to a variable or can be used within an `if` condition. So, we should only prohibit to use it without assigning to a variable or not using inside an `if` condition.

Reply:
Ok. Maybe you can achieve this with IntelliJ's structural search inspections: https://github.com/apache/incubator-druid/blob/78fd5aff21f692f583955a281e841b1e8fd6925f/ci/README_TeamCity.md#creating-a-custom-inspection-from-a-structural-search-pattern

Reply:
@kamaci could you please prohibit using Checkstyle like
```
^\s+Thread\.interrupted\(\)\;
```